src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -238,28 +239,38 @@ public DesiredCapabilities buildCapabilities(ConductorConfig config) {
     /**
      * Method that acts as an arbiter of implicit timeouts of sorts
      */
-    public WebElement waitForElement(String id) {
+    public MobileElement waitForElement(String id) {
         return waitForElement(PageUtil.buildBy(configuration, id));
     }
 
-    public WebElement waitForElement(String id, long timeoutInSeconds) {
+    public MobileElement waitForElement(String id, long timeoutInSeconds) {
         return waitForElement(PageUtil.buildBy(configuration, id), timeoutInSeconds);
     }
 
-    public WebElement waitForElement(By by) {
+    public MobileElement waitForElement(By by) {
         return waitForElement(by, configuration.getAppiumRequestTimeout());
     }
 
-    public WebElement waitForElement(By by, long timeoutInSeconds) {
+    public MobileElement waitForElement(By by, long timeoutInSeconds) {
         waitForCondition(ExpectedConditions.not(ExpectedConditions.invisibilityOfElementLocated(by)), timeoutInSeconds);
-        return getAppiumDriver().findElement(by);
+        return (MobileElement) getAppiumDriver().findElement(by);
     }
 
-    public WebElement waitForElement(WebElement element) {
+    @Deprecated
+    public MobileElement waitForElement(WebElement element) {
+        return waitForElement((MobileElement) element);
+    }
+
+    @Deprecated
+    public MobileElement waitForElement(WebElement element, long timeoutInSeconds) {
+        return waitForElement((MobileElement) element, timeoutInSeconds);
```

Comment:
So my thought was to have this the other way around, have the implementation here, and have the other method call this one. This is because _if_ the `WebElement` isn't a `MobileElement` you don't want to throw a class cast exception. It's always safe to upcast (`MobileElement` to `WebElement`). It's not always safe to downcast `MobileElement` to `WebElement`).

Reply:
Correct

Reply:
Is the only case that a `WebElement` isn't a `MobileElement` when automating native web views?

Reply:
@evant  Addressed in 1a33c9e